File: android/app/src/main/java/com/aura/voicechat/domain/model/Kyc.kt
package com.aura.voicechat.domain.model

/**
 * KYC (Know Your Customer) domain models
 * Developer: Hawkaye Visions LTD — Pakistan
 * 
 * IMPORTANT: Only ID Card (front/back) and Selfie verification
 * NO utility bills required
 */
data class KycData(
    val userId: String,
    val status: KycStatus,
    val idCardFront: String?,
    val idCardBack: String?,
    val selfie: String?,
    val livenessScore: Float?,
    val submittedAt: Long?,
    val reviewedAt: Long?,
    val rejectionReason: String?
)

data class KycSubmission(
    val idCardFrontUri: String,
    val idCardBackUri: String,
    val selfieUri: String,
    val livenessCheckPassed: Boolean
)

/**
 * KYC verification steps (NO utility bills)
 */
enum class KycStep {
    ID_CARD_FRONT,
    ID_CARD_BACK,
    SELFIE_CAPTURE,
    LIVENESS_CHECK,
    REVIEW_SUBMIT,
    COMPLETE
}

/**
 * Liveness detection result for selfie verification
 */
data class LivenessResult(
    val passed: Boolean,
    val faceDetected: Boolean,
    val eyesOpen: Boolean,
    val lookingStraight: Boolean,
    val score: Float
)

File: backend/src/routes/kyc.ts
/**
 * KYC Routes
 * Developer: Hawkaye Visions LTD — Pakistan
 * 
 * Only ID Card (front/back) + Selfie - NO utility bills
 */

import { Router } from 'express';
import { authenticate } from '../middleware/auth';
import { generalLimiter } from '../middleware/rateLimiter';
import * as kycController from '../controllers/kycController';

const router = Router();

// Get KYC status
router.get('/status', generalLimiter, authenticate, kycController.getKycStatus);

// Submit KYC
router.post('/submit', generalLimiter, authenticate, kycController.submitKyc);

// Get upload URL for ID card front
router.post('/upload/id-front', generalLimiter, authenticate, kycController.getIdFrontUploadUrl);

// Get upload URL for ID card back
router.post('/upload/id-back', generalLimiter, authenticate, kycController.getIdBackUploadUrl);

// Get upload URL for selfie
router.post('/upload/selfie', generalLimiter, authenticate, kycController.getSelfieUploadUrl);

export default router;
